theseus/embodied/misc/local_cost_fn.py

```diff
@@ -33,13 +34,31 @@ def __init__(
         self.register_optim_vars(["var"])
         self.register_aux_vars(["target"])
 
+        self._jac_cache: torch.Tensor = None
+
     def error(self) -> torch.Tensor:
         return self.target.local(self.var)
 
     def jacobians(self) -> Tuple[List[torch.Tensor], torch.Tensor]:
-        Jlist: List[torch.Tensor] = []
-        error = self.target.local(self.var, jacobians=Jlist)
-        return [Jlist[1]], error
+        if _THESEUS_GLOBAL_PARAMS.fast_approx_log_maps:
```

Comment:
If I remember correctly, is this the version used in MiniSAM?

Reply:
FYI, this one seems reasonable only if the `target` and `var` are close to each other. Not sure if we should implement it here.